osmoutils/osmocli/query_cmd_wrap.go

```diff
@@ -0,0 +1,166 @@
+package osmocli
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	grpc1 "github.com/gogo/protobuf/grpc"
+	"github.com/gogo/protobuf/proto"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+
+	"github.com/osmosis-labs/osmosis/v13/osmoutils"
+)
+
+func QueryIndexCmd(moduleName string) *cobra.Command {
+	return &cobra.Command{
+		Use:                        moduleName,
+		Short:                      fmt.Sprintf("Querying commands for the %s module", moduleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       indexRunCmd,
+	}
+}
+
+func indexRunCmd(cmd *cobra.Command, args []string) error {
+	usageTemplate := `Usage:{{if .HasAvailableSubCommands}}
+  {{.CommandPath}} [command]{{end}}
+  
+{{if .HasAvailableSubCommands}}Available Commands:{{range .Commands}}{{if .IsAvailableCommand}}
+  {{rpad .Name .NamePadding }} {{.Short}}{{end}}{{end}}
+
+Use "{{.CommandPath}} [command] --help" for more information about a command.{{end}}
+`
+	cmd.SetUsageTemplate(usageTemplate)
+	return cmd.Help()
+}
+
+type QueryDescriptor struct {
+	Use   string
+	Short string
+	Long  string
+
+	HasPagination bool
+
+	QueryFnName string
+}
+
+func SimpleQueryFromDescriptor[reqP proto.Message, querier any](desc QueryDescriptor, newQueryClientFn func(grpc1.ClientConn) querier) *cobra.Command {
+	numArgs := ParseNumFields[reqP]()
+	if desc.HasPagination {
+		numArgs = numArgs - 1
+	}
+	cmd := &cobra.Command{
+		Use:   desc.Use,
+		Short: desc.Short,
+		Long:  desc.Long,
+		Args:  cobra.ExactArgs(numArgs),
+		RunE: NewQueryLogicAllFieldsAsArgs[reqP](
+			desc.QueryFnName, newQueryClientFn),
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	if desc.HasPagination {
+		cmdName := strings.Split(desc.Use, " ")[0]
+		flags.AddPaginationFlagsToCmd(cmd, cmdName)
+	}
+
+	return cmd
+}
+
+// SimpleQueryCmd builds a query, for the common, simple case.
+// It detects that the querier function name is the same as the ProtoMessage name,
+// with just the "Query" and "Request" args chopped off.
+// It expects all proto fields to appear as arguments, in order.
```

Comment:
Do we always want *all* proto fields to be arguments? Some may have sensible defaults (if not timestamp is provided, use "now") or be overriden (e.g.: Sender in messages, which is forced to be the current sender).

Reply:
Looked at the rest of the code. I think this works great for now, but maybe in the future (when we have need for more customization) we could expand it using a builder pattern to manipulate certain args in client/query.go after applying this)

Reply:
Yeah, need to handle that for other messages. Thus far this is replacing everything that already follows this form.

Reply:
A builder pattern SGTM!